.travis.yml

```diff
@@ -107,9 +107,9 @@ jobs:
 
   include:
     - <<: *lint
-      name: linting the code
+      name: linting the code and extra checks
       env:
-        TOXENV: lint
+        TOXENV: lint,check
```

Comment:
Not really into this. The testenv should do one thing. Also in the description of the "check" it said it "should cover anything that is not linter" but then you included it in the linting environment?

Reply:
@decentral1se we cannot really afford to add more jobs to Travis. Each job adds a lot of extra delay because they are serialized, only ~3 run in parallel, the other are waiting in queue.

Reply:
Well, how about using this pre-commit then since actually this job does *a lot* of stuff and that is the whole point of pre-commit, in the end, no? We could adapt from https://github.com/szebenyib/pre-commit-pytest?

Reply:
Or pre-commit serves other more "lint" reasons ... ugh, it's all a big blur to me

Reply:
While there is always the question about why not running it in lint, I would like to keep the pre-commit run to the linting, Few reasons:
* pre-commit is the only dependency in that environment, as the tool has its own manged venvs
* more check will slowdown lint, and people will start skipping  it before push